src/main/java/com/cst438/controller/AssignmentController.java

```diff
@@ -17,102 +16,186 @@
 import java.util.ArrayList;
 import java.util.List;
 
+
+
 @RestController
 @CrossOrigin(origins = "http://localhost:3000")
 public class AssignmentController {
 
+    @Autowired
+    private AssignmentRepository assignmentRepository;
+    @Autowired
+    private SectionRepository sectionRepository;
+    @Autowired
+    private GradeRepository gradeRepository;
+    @Autowired
+    private EnrollmentRepository enrollmentRepository;
+
 
     // instructor lists assignments for a section.  Assignments ordered by due date.
     // logged in user must be the instructor for the section
+    // TEST URL http://localhost:8080/sections/8/assignments
     @GetMapping("/sections/{secNo}/assignments")
-    public List<AssignmentDTO> getAssignments(
-            @PathVariable("secNo") int secNo) {
-
-        // TODO remove the following line when done
-		
-		// hint: use the assignment repository method 
-		//  findBySectionNoOrderByDueDate to return 
-		//  a list of assignments
-
-        return null;
+    public List<AssignmentDTO> getAssignments(@PathVariable("secNo") int secNo) {
+
+        List<Assignment> assignments = assignmentRepository.findBySectionNoOrderByDueDate(secNo);
+
+        List<AssignmentDTO> assignmentDTOList = new ArrayList<>();
+        for (Assignment a : assignments) {
+            assignmentDTOList.add(new AssignmentDTO(
+                a.getAssignmentId(),
+                a.getTitle(),
+                a.getDueDate(),
+                a.getSection().getCourse().getCourseId(),
+                a.getSection().getSecId(),
+                a.getSection().getSectionNo()));
+        }
+        return assignmentDTOList;
     }
 
     // add assignment
     // user must be instructor of the section
     // return AssignmentDTO with assignmentID generated by database
+    //TEST URL http://localhost:8080/assignments
+    //TEST BODY {"title":"Assignment Post Test","dueDate":"2024-05-01","secId":1,"secNo":8}
     @PostMapping("/assignments")
     public AssignmentDTO createAssignment(
-            @RequestBody AssignmentDTO dto) {
-
-        // TODO remove the following line when done
-
-        return null;
-    }
-
-    // update assignment for a section.  Only title and dueDate may be changed.
-    // user must be instructor of the section
-    // return updated AssignmentDTO
-    @PutMapping("/assignments")
-    public AssignmentDTO updateAssignment(@RequestBody AssignmentDTO dto) {
-
-        // TODO remove the following line when done
-
-        return null;
+        @RequestBody AssignmentDTO assignmentDTO) {
+        Assignment a = new Assignment();
+        a.setAssignmentId(assignmentDTO.id());
```

Comment:
Don't think we need to add set Id's for entities that have auto generated primary key. assignmentDTO.id() should be null here.

Reply:
Good catch. As you said, Id's should be auto generated.